core/src/main/java/com/linecorp/armeria/client/retry/ExponentialBackoff.java

```diff
@@ -30,19 +30,16 @@
                       "initialDelayMillis: %s (expected: >= 0)", initialDelayMillis);
         checkArgument(initialDelayMillis <= maxDelayMillis, "maxDelayMillis: %s (expected: >= %s)",
                       maxDelayMillis, initialDelayMillis);
-        currentDelayMillis = initialDelayMillis;
+        this.initialDelayMillis = initialDelayMillis;
         this.maxDelayMillis = maxDelayMillis;
         this.multiplier = multiplier;
     }
 
     @Override
     protected long doNextDelayMillis(int numAttemptsSoFar) {
-        if (currentDelayMillis >= maxDelayMillis) {
-            return maxDelayMillis;
-        }
-        long nextDelay = currentDelayMillis;
-        currentDelayMillis = saturatedMultiply(currentDelayMillis, multiplier);
-        return nextDelay;
+        final long nextDelay =
+                saturatedMultiply(initialDelayMillis, Math.pow(multiplier, numAttemptsSoFar - 1));
+        return nextDelay >= maxDelayMillis ? maxDelayMillis : nextDelay;
```

Comment:
nit: How about using `Math.min(nextDelay, maxDelayMillis)` to easily understand?